src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5536,44 +5537,52 @@ private NameSyntax ParseQualifiedName(NameOptions options = NameOptions.None)
         {
             var right = this.ParseSimpleName(options);
 
-            if (separator.Kind == SyntaxKind.DotToken)
+            switch (separator.Kind)
             {
-                return _syntaxFactory.QualifiedName(left, separator, right);
-            }
-            else if (separator.Kind == SyntaxKind.ColonColonToken)
-            {
-                if (left.Kind != SyntaxKind.IdentifierName)
-                {
-                    separator = this.AddError(separator, ErrorCode.ERR_UnexpectedAliasedName, separator.ToString());
-                }
+                case SyntaxKind.DotToken:
+                    return _syntaxFactory.QualifiedName(left, separator, right);
+                case SyntaxKind.DotDotToken:
+                    // Error recovery.  If we have `X..Y` break that into `X.<missing-id>.Y`
 
-                // If the left hand side is not an identifier name then the user has done
-                // something like Goo.Bar::Blah. We've already made an error node for the
-                // ::, so just pretend that they typed Goo.Bar.Blah and continue on.
+                    var leftDot = SyntaxFactory.Token(separator.LeadingTrivia.Node, SyntaxKind.DotToken, null);
+                    var rightDot = this.AddError(SyntaxFactory.Token(null, SyntaxKind.DotToken, separator.TrailingTrivia.Node), ErrorCode.ERR_IdentifierExpected);
```

Comment:
>ErrorCode.ERR_IdentifierExpected [](start = 128, length = 32)

I would expect the error on the missing identifier instead

Reply:
Done.